lib/icon/module.js

```diff
@@ -113,7 +114,8 @@ async function generateIcons (options) {
   options._assetIcons = options.sizes.map(size => ({
     src: joinUrl(options.publicPath, options._icons[size]),
     sizes: `${size}x${size}`,
-    type: 'image/png'
+    type: 'image/png',
+    purpose: options.purpose ? options.purpose.join(' ') : undefined
```

Comment:
- Add validation, the types allowed: `'badge', 'maskable', 'any'`
- Also recommend checking if `purpose` is string, put in array